modules/os2forms_digital_signature/src/Service/SigningService.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+namespace Drupal\os2forms_digital_signature\Service;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Config\ImmutableConfig;
+use Drupal\Core\Entity\EntityStorageInterface;
+use Drupal\Core\Entity\EntityTypeManager;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Drupal\os2forms_digital_signature\Form\SettingsForm;
+use GuzzleHttp\ClientInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+/**
+ * Digital signing service.
+ */
+class SigningService {
+
+  /**
+   * The config.
+   *
+   * @var \Drupal\Core\Config\ImmutableConfig
+   */
+  private readonly ImmutableConfig $config;
+
+  /**
+   * Webform storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected EntityStorageInterface $webformStorage;
+
+  /**
+   * WebformSubmission storage.
+   *
+   * @var \Drupal\Core\Entity\EntityStorageInterface
+   */
+  protected EntityStorageInterface $webformSubmissionStorage;
+
+  public function __construct(
+    private readonly ClientInterface $httpClient,
+    private readonly TimeInterface $time,
+    ConfigFactoryInterface $configFactory,
+    EntityTypeManager $entityTypeManager,
+    private readonly LoggerChannelInterface $logger,
+  ) {
+    $this->config = $configFactory->get(SettingsForm::$configName);
+    $this->webformStorage = $entityTypeManager->getStorage('webform');
+    $this->webformSubmissionStorage = $entityTypeManager->getStorage('webform_submission');
+  }
+
+  /**
+   * Fetch a new cid.
+   *
+   * @return string|null
+   *   The correlation id.
+   */
+  public function getCid() : ?string {
+    $url = $this->config->get('os2forms_digital_signature_remove_service_url') . 'action=getcid';
```

Comment:
This will break if "os2forms_digital_signature_remove_service_url" does not end with `?` or `&`. Use `http_build_query` like you do in two other places in this class. Adding a helper function, `getServiceUrl`, say, would be a good idea since you need this URL in multiple places. This function should also handle the case when the base URL already contains `?` or even `&`.

Reply:
@stankut, please address this comment.

Reply:
Sorry, @stankut, GitHub confused me. You've already addressed this.